services/web/server/tests/unit/with_dbs/01/test_api_keys.py

```diff
@@ -172,9 +171,15 @@ async def test_create_api_key_with_expiration(
         assert [d["displayName"] for d in data] == ["foo"]
 
         # wait for api-key for it to expire and force-run scheduled task
-        await asyncio.sleep(expiration_interval.seconds)
-        deleted = await prune_expired_api_keys(client.app)
-        assert deleted == ["foo"]
+        async for attempt in tenacity.AsyncRetrying(
+            wait=wait_fixed(1),
+            retry=retry_if_exception_type(AssertionError),
+            stop=stop_after_attempt(5),
```

Comment:
Q: can you define an upper bound based on `expiration_interval.seconds` for this number? 
NOTE that there are many different stop/wait policies (not only based on number of attempts and/or time!)